lib/connectors_sdk/office365/extractor.rb

```diff
@@ -61,14 +51,17 @@ def yield_document_changes(modified_since: nil, break_after_page: false, &block)
             capture_exception(e)
           end
 
-          if break_after_page
-            if config.cursors['page_cursor'].blank?
-              config.cursors['last_drive_id'] = config.cursors.delete('current_drive_id')
-            end
+          if break_after_page && config.cursors['page_cursor'].present?
             break
           end
         end
 
+        if break_after_page && config.cursors['page_cursor'].blank?
+          @completed = true
+          config.overwrite_cursors!(retrieve_latest_cursors)
```

Comment:
@wangch079 I think at least one issue with the _full_ sync is with this line. You don't want to record those cursors, which are only used by _incremental_ syncs. It's possible they get passed on to the next full sync, making it behave like an incremental sync?

Reply:
It won't. `cursors` will always be `nil` for non-incremental syncs, check [here](https://github.com/elastic/ent-search/blob/7286805237716d62a5cdf0d9c1014d32a71b2ae8/frito_pie/app/models/frito_pie/content_source.rb#L651-L658)

The reason that `cursors` are persisted for `full` sync is because we try to retrieve `cursors` from the last completed job if we can't find the last completed `incremental` job ([ref](https://github.com/elastic/ent-search/blob/7286805237716d62a5cdf0d9c1014d32a71b2ae8/frito_pie/app/models/frito_pie/content_source.rb#L652)):
```
last_job = most_recent_completed_job(Connectors::Job::INCREMENTAL) || most_recent_completed_job
```

We actually persist `cursors` for both [full/incremental](https://github.com/elastic/ent-search/blob/7286805237716d62a5cdf0d9c1014d32a71b2ae8/connectors/lib/connectors/work/index.rb#L65-L68) job and [delete](https://github.com/elastic/ent-search/blob/7286805237716d62a5cdf0d9c1014d32a71b2ae8/connectors/lib/connectors/work/purge.rb#L59-L62) job. We don't do the same for [permissions](https://github.com/elastic/ent-search/blob/7286805237716d62a5cdf0d9c1014d32a71b2ae8/connectors/lib/connectors/work/permissions.rb#L36-L38) job, I guess we just missed it.

Reply:
Got it, that makes sense 👍